src/librustdoc/html/static/main.js

```diff
@@ -37,10 +37,58 @@
                      "associatedconstant",
                      "union"];
 
+    function hasClass(elem, className) {
+        if (elem && className && elem.className) {
+            var x = elem.className.split(' ');
+            for (var i = 0; i < x.length; i++) {
+                if (x[i] === className) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    function addClass(elem, className) {
+        if (elem && className && !hasClass(elem, className)) {
+            if (elem.className && elem.className.length > 0) {
+                elem.className += ' ' + className;
+            } else {
+                elem.className = className;
+            }
+        }
+    }
+
+    function removeClass(elem, className) {
+        if (elem && className && elem.className) {
+            var x = elem.className.split(' ');
+            for (var i = 0; i < x.length; i++) {
+                if (x[i] === className) {
+                    x.splice(i, 1);
+                }
+            }
+            elem.className = x.join(' ');
+        }
+    }
+
+    function on_each(arr, func) {
```

Comment:
@isonmad Really depends on browsers you are targetting, but `Array.from` is supported only by recent browseres, i.e. not IE. And if that's of no concern, you could also [use `for`/`of`](http://stackoverflow.com/a/22754453/240963).

You could use `forEach` here, although it is generally less performant:

```js
Array.prototype.forEach.call(arr, func)
```